File: cpp/sorting_algorithms/bubble_sort.cpp
#include<iostream>
using namespace std;

// Time Complexity O(n^2)

void bubble_sort(int A[], int n) {
    for (int k = 1; k <= n - 1; k++) {
        bool swap = false;

        for (int i = 0; i <= n - k - 1; i++) {
            if(A[i] > A[i + 1]) {
                int temp = A[i];
                A[i] = A[i + 1];
                A[i + 1] = temp;

                swap = true;
            }
        }

        if(!swap) {
            break;
        }
    }
}

int main () {
    int A[] = {2, 7, 4, 1, 5, 3};
    bubble_sort(A, 6);

    cout << "Sorted Array: ";
    for (int i = 0; i < 6; i++) {
        cout << A[i] << " ";
    }

    return 0;
}

File: javascript/sortingAlgorithms/bubbleSort.js
// Bubbles the largest value to the end
// Time Complexity - O(n^2)
// Space Complexity - O(1)

function bubbleSort(arr) {
    for (let i = arr.length - 1; i > 0; i--) {
        let swap = false;

        for(let j = 0; j < i; j++) {
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swap = true;
            }
        }

        if(!swap) break;
    }

    return arr;
}